htdocs/accountancy/class/accountingaccount.class.php

```diff
@@ -868,20 +868,44 @@ public function getAccountingCodeToBind(Societe $buyer, Societe $seller, Product
 			}
 
 			// Manage Deposit
-			if ($factureDet->desc == "(DEPOSIT)" || $facture->type == $facture::TYPE_DEPOSIT) {
-				$accountdeposittoventilated = new self($this->db);
-				if ($type == 'customer') {
-					$result = $accountdeposittoventilated->fetch('', $conf->global->ACCOUNTING_ACCOUNT_CUSTOMER_DEPOSIT, 1);
-				} elseif ($type == 'supplier') {
-					$result = $accountdeposittoventilated->fetch('', $conf->global->ACCOUNTING_ACCOUNT_SUPPLIER_DEPOSIT, 1);
-				}
-				if (isset($result) && $result < 0) {
-					return -1;
+			if (!empty($conf->global->{'ACCOUNTING_ACCOUNT_' . strtoupper($type) . '_DEPOSIT'})) {
```

Comment:
Syntax $conf->global->{  is forbidden. Result differs with PHP version.
Use instead   getDolGlobalString( )